src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -129,16 +149,34 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
     prev_index: input.outputIndex,
   })
 
+  const prepareTokenBundle = (
+    multiAssets: _MultiAsset[],
+  ): TrezorMultiAsset | undefined => {
+    const tokenBundle = multiAssets.map(({ policyId, assets }) => {
+      const tokenAmounts = assets.map(({ assetName, amount }) => ({
+        assetNameBytes: assetName.toString('hex'),
+        amount: amount.toString(),
+      }))
+      return {
+        policyId: policyId.toString('hex'),
+        tokenAmounts,
+      }
+    })
+    return tokenBundle.length > 0 ? tokenBundle : undefined
```

Comment:
I think trezor-connect allows an empty array of tokens to be passed, we don't need to resort to returning `undefined` and even if we did, `null` would be more suitable as `undefined` may seem like some unhandled scenario in the code which it really isn't

Reply:
Fixed